### PR TITLE
Add cron_enabled attribute to good_job and pass it to process current state

### PR DESCRIPTION
**Pull request by saksham-jain**
- Add cron_enabled attribute to good_job.
- Set this attribute on initialization using enable_cron key in configuration.
- Add new key cron_enabled to process current_state and use GoodJob.cron_enabled to set this key.

### CODE REVIEW

File: lib/good_job.rb [hunk 1]
```diff
@@ -59,6 +59,14 @@ module GoodJob
   #   @return [Boolean, nil]
   mattr_accessor :retry_on_unhandled_error, default: false
 
+  # @!attribute [rw] cron_enabled
+  #   @!scope class
+  #   Use this to know if cron is enabled for the process (default: +false+).
+  #   If +true+, it means cron is enabled.
+  #   If +false+, it means cron disabled.
+  #   @return [Boolean, nil]
+  mattr_accessor :cron_enabled, default: false
```
Comment (bensheldon) [inline lib/good_job.rb:68]: I'm trying to move away from these global accessors.

Take a look at `GoodJob::Configuration` and see if you can instantiate one of those instead to get the configuration.

**Review by bensheldon — CHANGES_REQUESTED** [review]

File: lib/good_job.rb (hunk 1 above)
Comment (saksham-jain) [reply lib/good_job.rb:68]: Should we directly add new key in process's current_state without creating this global accessor, like this -> `cron_enabled: Rails.application.config.good_job[:enable_cron]`

Comment (bensheldon) [reply lib/good_job.rb:68]: I'm working on #681, which should allow us to use `GoodJob.configuration.enable_cron?`. I'll let you know when it gets merged.

Comment (bensheldon) [reply lib/good_job.rb:68]: It's been merged. If you merge main or rebase, you can use `GoodJob.configuration`

Comment (saksham-jain) [reply lib/good_job.rb]: Okay, This made things easy.

**Comment by bensheldon** [comment]
@saksham-jain could you please fix up the lint warning? This looks good 👍🏻

**Review by bensheldon — APPROVED** [review]
👍🏻 thank you!

**Comment by bensheldon** [comment]
@saksham-jain Thank you for the PR. I really appreciate this! Are you interested in further issues?

Just having worked through with this, I see a few jumping off points for improvement:

- Rename that configuration option to be `Configuration#cron_enabled?` so it's consistent and the less awkward construction.
- Could you go through `Configuration` and propose other configuration options you think would be meaningful to expose on the process state for debugging in the dashboard?

**Comment by saksham-jain** [comment]
Thanks, @bensheldon. I will check these points.

**Pull request merged**